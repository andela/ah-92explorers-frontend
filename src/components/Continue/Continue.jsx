import React, { Component, Fragment } from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { Button } from 'reactstrap';
import { getInfo } from '../../redux/actions/actionCreators';

export class Continue extends Component {
  handleOnClick = () => {
    const searchURL = window.location.search;
    this.props.getInfo(searchURL);
  }

  render() {
    const { success } = this.props.user;
    return (
      <Fragment>
        <div className="row continue">
          <div className="col-sm-12">
            <div className="text-center">
              <Button className="btn__continue" onClick={this.handleOnClick}>Continue to Authors Haven </Button>
              {
                success && <Redirect to="/" />
              }
            </div>
          </div>
        </div>
      </Fragment>
    );
  }
}

const mapStateToProps = (state) => ({
  user: state.social,
});

const socialConnected = connect(mapStateToProps, { getInfo })(Continue);
export { socialConnected as Continuee };
