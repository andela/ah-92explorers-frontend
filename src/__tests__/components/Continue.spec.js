import React from 'react';
import { shallow, mount } from 'enzyme';
import { Continue, mapStateToProps } from '../../components/Continue/Continue.jsx';

describe('<Continue />', () => {
    const props = {
        getInfo: jest.fn(),
        user: { success:{} }
    };
  
    const wrapper = shallow(<Continue {...props}/>);
  
    it('should render Comments component', () => {
      expect(wrapper).toMatchSnapshot();
    });
  
    it('should render one main Fragment element', () => {
      expect(wrapper.find('Fragment')).toHaveLength(1);
    });
  
    it('should handle on click', () => {
      const component = wrapper.instance();
      component.handleOnClick();
      expect(component).toBeDefined();
    });

    it('to have wrapper class', async () => {
      const wrapper = shallow(<Continue {...props} />);
      const btn = wrapper.find('.btn__continue');
      btn.simulate('click');
      expect(btn.length).toBe(1);
    });

    it('should map state to props', () => {
        const initialState = {
          social: {
            user: { success: {}},
          }
        };
        mapStateToProps({...initialState});
        expect(mapStateToProps).toBe(mapStateToProps);
      });
  });