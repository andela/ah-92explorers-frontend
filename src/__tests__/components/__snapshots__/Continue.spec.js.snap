// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Continue /> should render Comments component 1`] = `ShallowWrapper {}`;
